File: sohee-jung/week3/백준_silver_DFS와 BFS.java
import java.io.*;
import java.util.LinkedList;
import java.util.Queue;
import java.util.StringTokenizer;

public class Main {

    static int N;
    static int M;
    static int[][] map;
    static boolean[] visitDfs;
    static boolean[] visitBfs;

    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        BufferedWriter bw = new BufferedWriter(new OutputStreamWriter(System.out));

        StringTokenizer st = new StringTokenizer(br.readLine());
        N = Integer.parseInt(st.nextToken());
        M = Integer.parseInt(st.nextToken());
        int V = Integer.parseInt(st.nextToken());

        map = new int[N + 1][N + 1];
        visitDfs = new boolean[N + 1];
        visitBfs = new boolean[N + 1];

        for (int i = 0; i < M; i++) {
            st = new StringTokenizer(br.readLine());
            int firstVertex = Integer.parseInt(st.nextToken());
            int secondVertex = Integer.parseInt(st.nextToken());

            map[firstVertex][secondVertex] = 1;
            map[secondVertex][firstVertex] = 1;
        }

        dfs(V, bw);
        bw.write("\n");
        bfs(V, bw);

        bw.flush();
        br.close();
        bw.close();
    }

    public static void dfs(int start, BufferedWriter bw) throws IOException {
        visitDfs[start] = true;
        bw.write(start + " ");

        for (int i = 1; i <= N; i++) {
            if (map[start][i] == 1 && !visitDfs[i]) {
                dfs(i, bw);
            }
        }
    }

    public static void bfs(int start, BufferedWriter bw) throws IOException {
        Queue<Integer> queue = new LinkedList<>();
        queue.add(start);

        visitBfs[start] = true;

        while (!queue.isEmpty()){
            int current = queue.poll();
            bw.write(current + " ");

            for (int i = 1; i <= N; i++) {
                if (map[current][i] == 1 && !visitBfs[i]){
                    queue.add(i);
                    visitBfs[i] = true;
                }

            }
        }
    }
}

File: minseo-song/week3/백준_silver_DFS와 BFS.kt
package week3

import java.io.BufferedReader
import java.io.BufferedWriter
import java.io.InputStreamReader
import java.io.OutputStreamWriter
import java.util.LinkedList
import java.util.Queue
import java.util.StringTokenizer

fun main() {
    val bufferedReader = BufferedReader(InputStreamReader(System.`in`))
    val bufferedWriter = BufferedWriter(OutputStreamWriter(System.out))

    val st = StringTokenizer(bufferedReader.readLine())
    val n = st.nextToken().toInt()
    val m = st.nextToken().toInt()
    val t = st.nextToken().toInt()

    val graph = Array(n + 1) { IntArray(n + 1) }

    repeat(m) {
        val edge = StringTokenizer(bufferedReader.readLine())
        val a = edge.nextToken().toInt()
        val b = edge.nextToken().toInt()

        graph[a][b] = 1
        graph[b][a] = 1
    }

    fun dfs(graph: Array<IntArray>, visited: BooleanArray, v: Int, writer: BufferedWriter) {
        visited[v] = true
        writer.write("$v ")

        for (i in 1 until graph.size) {
            if (!visited[i] && graph[v][i] == 1) {
                dfs(graph, visited, i, writer)
            }
        }
    }

    fun bfs(graph: Array<IntArray>, visited: BooleanArray, start: Int, writer: BufferedWriter) {
        val queue: Queue<Int> = LinkedList()
        queue.add(start)
        visited[start] = true

        while (queue.isNotEmpty()) {
            val v = queue.poll()
            writer.write("$v ")

            for (i in 1 until graph.size) {
                if (!visited[i] && graph[v][i] == 1) {
                    queue.add(i)
                    visited[i] = true
                }
            }
        }
    }

    val visitedDfs = BooleanArray(n+1)
    dfs(graph, visitedDfs, t, bufferedWriter)
    bufferedWriter.write("\n")

    val visitedBfs = BooleanArray(n+1)
    bfs(graph, visitedBfs, t, bufferedWriter)
    bufferedWriter.close()
}